src/client/client.go

```diff
@@ -296,59 +297,62 @@ func getCertOptionsFromEnv() ([]Option, error) {
 // getUserMachineAddrAndOpts is a helper for NewOnUserMachine that uses
 // environment variables, config files, etc to figure out which address a user
 // running a command should connect to.
-func getUserMachineAddrAndOpts(context *config.Context) (string, []Option, error) {
+func getUserMachineAddrAndOpts(context *config.Context) (*grpcutil.PachdAddress, []Option, error) {
 	var options []Option
 
 	// 1) PACHD_ADDRESS environment variable (shell-local) overrides global config
-	if envAddr, ok := os.LookupEnv("PACHD_ADDRESS"); ok {
+	if envAddrStr, ok := os.LookupEnv("PACHD_ADDRESS"); ok {
 		fmt.Fprintln(os.Stderr, "WARNING: 'PACHD_ADDRESS' is deprecated and will be removed in a future release, use Pachyderm contexts instead.")
 
-		if strings.HasPrefix(envAddr, grpcs) {
-			options = append(options, WithSystemCAs)
-			envAddr = strings.TrimPrefix(envAddr, grpcs)
+		envAddr, err := grpcutil.ParsePachdAddress(envAddrStr)
+		if err != nil {
+			return nil, nil, fmt.Errorf("could not parse 'PACHD_ADDRESS': %v", err)
 		}
-		if !strings.Contains(envAddr, ":") {
-			envAddr = fmt.Sprintf("%s:%s", envAddr, DefaultPachdNodePort) // append port
+
+		if envAddr.Secured {
+			options = append(options, WithSystemCAs)
 		}
+
 		options, err := getCertOptionsFromEnv()
 		if err != nil {
-			return "", nil, err
+			return nil, nil, err
 		}
-		return envAddr, options, nil
+
+		return &envAddr, options, nil
 	}
 
 	// 2) Get target address from global config if possible
 	if context != nil && (context.ServerCAs != "" || context.PachdAddress != "") {
+		pachdAddress, err := grpcutil.ParsePachdAddress(context.PachdAddress)
+		if err != nil {
+			return nil, nil, fmt.Errorf("could not parse the active context's pachd address: %v", err)
+		}
+
 		// Proactively return an error in this case, instead of falling back to the default address below
-		if context.ServerCAs != "" && !strings.HasPrefix(context.PachdAddress, grpcs) {
-			return "", nil, fmt.Errorf("must set pachd_address to %s... if server_cas is set", grpcs)
+		if context.ServerCAs != "" && !pachdAddress.Secured {
+			return nil, nil, errors.New("must set pachd_address to grpc://... if server_cas is set")
```

Comment:
I think `!pachdAddress.Secured` implies that the prefix is probably `grpc://` already (in other words, I think the error really should be `must set pachd address to grpcs://...`)